internal/rbd/controllerserver.go

```diff
@@ -1092,6 +1091,7 @@ func (cs *ControllerServer) CreateSnapshot(
 
 		return nil, err
 	}
+	defer rbdVol.Destroy()
```

Comment:
This can cause a cleanup problem as rbdVol is returned even in case of error. IMO moving this to line 1081 with a check for nil is okay.

Reply:
That's what I thought from the beginning

Reply:
`rbdVol` should never be returned in a case of an error. Either `err != nil` or `rbdVol != nil`. The additional if-statement should be useless, and in effect is the same as having the `defer` below the error handling. It adds a bit of complexity, and makes me wonder if `rbdVol == nil` should not be handled further down in the function as well?